diff --git a/src/audio/alsa/SDL_alsa_audio.c b/src/audio/alsa/SDL_alsa_audio.c
index f10733e4..139749df 100644
--- a/src/audio/alsa/SDL_alsa_audio.c
+++ b/src/audio/alsa/SDL_alsa_audio.c
@@ -179,7 +179,7 @@
 			device = "plug:surround40";
 			break;
 		default:
-			device = "default";
+			device = "plughw:Z2,0,0";
 			break;
 		}
 	}
@@ -437,6 +437,10 @@
 	snd_pcm_hw_params_t *hwparams;
 	snd_pcm_uframes_t frames;
 
++	/* Switch to blocking mode for playback */
++	/* Note: this must happen before hw/sw params are set. */
++	SDL_NAME(snd_pcm_nonblock)(pcm_handle, 0);
+
 	/* Copy the hardware parameters for this setup */
 	snd_pcm_hw_params_alloca(&hwparams);
 	SDL_NAME(snd_pcm_hw_params_copy)(hwparams, params);
@@ -611,9 +615,6 @@
 	}
 	SDL_memset(mixbuf, spec->silence, spec->size);
 
-	/* Switch to blocking mode for playback */
-	SDL_NAME(snd_pcm_nonblock)(pcm_handle, 0);
-
 	/* We're ready to rock and roll. :-) */
 	return(0);
 }
